analyses/transcriptomic-dimension-reduction/README.md

```diff
@@ -0,0 +1,130 @@
+# Dimension reduction: gene expression data
+
+This module runs dimension reduction—namely PCA, UMAP, and t-SNE—on gene expression data from kallisto and RSEM.
+
+To the run the whole pipeline and produce 4 plots with panels for all three approaches
+
+1. RSEM stranded data
+2. RSEM poly-A data
+3. kallisto stranded data
+4. kallisto poly-A data
+
+with points colored by broad histology, simply run:
+
+```
+./dimension-reduction-plots.sh
+```
+
+## Notes on further applications
+
+To document this module beyond what is in the individual files, we list some example applications or extensions below.
+
+Running dimension reduction, creating lists of plots, and generating the multipanel plots can all be run separately using the appropriate Rscripts in `scripts/`:
+
+```
+generate-multipanel-plot.R
+get-plot-list.R
+run-dimension-reduction.R
+```
+
+### Repeat dimension reduction step with different seeds
+
+##### Command line
+
+If we wanted to run UMAP multiple times with different seed values on the kallisto poly-A data, but _skip t-SNE_, we could do the following:
+
+```sh
+seeds=(309 4642 2019 2994 5783)
+
+for seed in "${seeds[@]}"
+do
+  Rscript --vanilla scripts/run-dimension-reduction.R \
+    --expression ../../data/pbta-gene-expression-kallisto.polya.rds \
+    --metadata ../../data/pbta-histologies.tsv \
+    --filename_lead kallisto_polyA_${seed} \
+    --output_directory results \
+    --seed ${seed} \
+    --skip_tsne
+done
+
+```
+
+Unfortunately, this approach will also run PCA 5 times. 
```

Comment:
Should it have an option to not do this?

Reply:
I think that use case is covered in the R section below. The way to do this would be to include logic to skip each of the PCA, UMAP, and t-SNE (or enable each step depending on how you did it), which is easy enough to do if it looks like it will come up often. I think a more likely scenario is that someone picked one of the three methods for display and is running it multiple times or tweaking the plots.